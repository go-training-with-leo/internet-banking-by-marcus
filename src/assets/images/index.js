import { ReactComponent as PlusIcon } from 'assets/images/icons/plus.svg';
import { ReactComponent as CreditCard } from 'assets/images/icons/creditCard.svg';
import { ReactComponent as Contact } from 'assets/images/icons/contact.svg';
import { ReactComponent as CreditCardDone } from 'assets/images/icons/creditCardDone.svg';
import { ReactComponent as Clock } from 'assets/images/icons/time.svg';
import { ReactComponent as DeleteIcon } from 'assets/images/icons/delete.svg';
import { ReactComponent as Edit } from 'assets/images/icons/edit.svg';
import { ReactComponent as Transfer } from 'assets/images/icons/transfer.svg';
import { ReactComponent as Cash } from 'assets/images/icons/cash.svg';
import { ReactComponent as CashAdd } from 'assets/images/icons/cashAdd.svg';
import { ReactComponent as Cancel } from 'assets/images/icons/cancel.svg';
import { ReactComponent as User } from 'assets/images/icons/user.svg';
import { ReactComponent as ArrowDown } from 'assets/images/icons/arrowDown.svg';
import { ReactComponent as Check } from 'assets/images/icons/check.svg';
import { ReactComponent as Reverse } from 'assets/images/icons/reverse.svg';
import { ReactComponent as NotifyBusy } from 'assets/images/icons/notificationBusy.svg';
import { ReactComponent as Info } from 'assets/images/icons/info.svg';
import { ReactComponent as NotifyFree } from 'assets/images/icons/notificationFree.svg';
import { ReactComponent as NotifyRemove } from 'assets/images/icons/notificationRemove.svg';
import { ReactComponent as Alert } from 'assets/images/icons/alert.svg';
import { ReactComponent as AlertYellow } from 'assets/images/icons/alertYellow.svg';
import { ReactComponent as Locker } from 'assets/images/icons/locker.svg';
import { ReactComponent as EightLogo } from 'assets/images/icons/eightBank.svg';
import { ReactComponent as MasterCard } from 'assets/images/icons/masterCard.svg';
import { ReactComponent as Next } from 'assets/images/icons/next.svg';
import { ReactComponent as Back } from 'assets/images/icons/back.svg';
import { ReactComponent as Ellipse } from 'assets/images/icons/ellipse.svg';
import { ReactComponent as Search } from 'assets/images/icons/search.svg';

import { ReactComponent as Loading } from 'assets/images/icons/loadingV2.svg';
import { ReactComponent as Visa } from 'assets/images/icons/visa.svg';
import { ReactComponent as Napas } from 'assets/images/icons/napas.svg';

import Eye from 'assets/images/icons/eye.png';
import EyeCross from 'assets/images/icons/eyeCross.png';

import { ReactComponent as ACB } from 'assets/images/icons/acb.svg';

import EightGif from './eight.gif';

export {
  ACB,
  Alert,
  AlertYellow,
  ArrowDown,
  Back,
  Cancel,
  Cash,
  CashAdd,
  Check,
  Clock,
  Contact,
  CreditCard,
  CreditCardDone,
  DeleteIcon,
  Edit,
  EightGif,
  EightLogo,
  Ellipse,
  Eye,
  EyeCross,
  Info,
  Loading,
  Locker,
  MasterCard,
  Napas,
  Next,
  NotifyBusy,
  NotifyFree,
  NotifyRemove,
  PlusIcon,
  Reverse,
  Search,
  Transfer,
  User,
  Visa,
};
