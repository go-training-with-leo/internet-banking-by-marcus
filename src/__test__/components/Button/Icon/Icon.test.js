/* eslint-disable no-console */
import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import IconButton from 'components/Button/Icon';
import { PlusIcon } from 'assets/images';

beforeAll(() => {
  jest.spyOn(console, 'error').mockImplementation(() => {});
});

test('Test default props button', () => {
  const handleClick = jest.fn();

  const tree = render(
    <IconButton onClick={handleClick}>
      Label
      <PlusIcon />
    </IconButton>
  );

  const button = tree.getByRole('button');

  expect(button.className).toMatch('button');

  expect(button).toHaveTextContent('Label');

  expect(button.childElementCount).toEqual(1);

  expect(console.error).toBeCalledTimes(1);

  fireEvent.click(button);

  expect(tree).toMatchSnapshot();
});

test('Test classname on Button', () => {
  const tree = render(<IconButton className='testClass'>Label</IconButton>);
  const button = tree.getByRole('button');

  expect(button.className).not.toMatch('testClass');

  expect(button.className).toMatch('button');

  expect(tree).toMatchSnapshot();
});

test('Wrong Props & right props', () => {
  const tree = render(
    <IconButton className='ghost' danger disabled>
      Label
    </IconButton>
  );

  const button = tree.getByRole('button');

  expect(button).not.toHaveAttribute('disabled');
  expect(button).not.toHaveClass('ghost');

  expect(button).toHaveAttribute('type', 'button');

  expect(tree).toMatchSnapshot();
});

test('Loading in button', () => {
  const tree = render(<IconButton loading>Label</IconButton>);

  const button = tree.getByRole('button');

  expect(button).toHaveAttribute('disabled');
  expect(tree).toMatchSnapshot();
});

test('Test onClick on Button', () => {
  const handleClick = jest.fn();
  const tree = render(
    <IconButton onClick={handleClick}>Test onClick</IconButton>
  );

  const button = tree.getByRole('button');

  fireEvent.click(button);

  expect(handleClick).toHaveBeenCalledTimes(1);

  expect(tree).toMatchSnapshot();
});
