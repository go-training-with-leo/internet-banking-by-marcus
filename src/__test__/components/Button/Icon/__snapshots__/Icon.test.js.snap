// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading in button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="button loading"
        disabled=""
        type="button"
      >
        <svg
          class="loader medium large"
        >
          loadingV2.svg
        </svg>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="button loading"
      disabled=""
      type="button"
    >
      <svg
        class="loader medium large"
      >
        loadingV2.svg
      </svg>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test classname on Button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="button"
        type="button"
      >
        Label
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="button"
      type="button"
    >
      Label
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test default props button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="button"
        type="button"
      >
        Label
        <svg>
          plus.svg
        </svg>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="button"
      type="button"
    >
      Label
      <svg>
        plus.svg
      </svg>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test onClick on Button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="button"
        type="button"
      >
        Test onClick
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="button"
      type="button"
    >
      Test onClick
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Wrong Props & right props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="button danger"
        type="button"
      >
        Label
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="button danger"
      type="button"
    >
      Label
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
